pyart/correct/bias_and_noise.py

```diff
@@ -0,0 +1,155 @@
+
+    correct_noise_rhohv
+    correct_bias
+
+"""
+
+import numpy as np
+
+from ..config import get_metadata, get_field_name, get_fillvalue
+
+
+def correct_noise_rhohv(radar, urhohv_field=None, snr_field=None,
+                        zdr_field=None, nh_field=None, nv_field=None,
+                        rhohv_field=None):
+    """
+    Corrects RhoHV for noise according to eq. 6 in Gourley et al. 2006.
+    This correction should only be performed if noise has not been subtracted
+    from the signal during the moments computation.
+
+    Parameters
+    ----------
+    radar : Radar
+        radar object
+
+    urhohv_field : str
+        name of the RhoHV uncorrected for noise field
+
+    snr_field, zdr_field, nh_field, nv_field: str
+        names of the SNR, ZDR, horizontal channel noise in dBZ and vertical
+        channel noise in dBZ used to correct RhoHV
+
+    rhohv_field: str
+        name of the rhohv field to output
+
+    Returns
+    -------
+    rhohv : dict
+        noise corrected RhoHV field
+
+    References
+    ----------
+    Gourley et al. Data Quality of the Meteo-France C-Band Polarimetric
+    Radar, JAOT, 23, 1340-1356
+
+    """
+    # parse the field parameters
+    if urhohv_field is None:
+        urhohv_field = get_field_name('uncorrected_cross_correlation_ratio')
+    if snr_field is None:
+        snr_field = get_field_name('signal_to_noise_ratio')
+    if zdr_field is None:
+        zdr_field = get_field_name('differential_reflectivity')
+    if nh_field is None:
+        nh_field = get_field_name('noisedBZ_hh')
+    if nv_field is None:
+        nv_field = get_field_name('noisedBZ_vv')
+    if rhohv_field is None:
+        rhohv_field = get_field_name('cross_correlation_ratio')
+
+    # extract fields from radar
+    if urhohv_field in radar.fields:
+        urhohv = radar.fields[urhohv_field]['data']
+    else:
+        raise KeyError('Field not available: ' + urhohv_field)
+    if snr_field in radar.fields:
+        snrdB_h = radar.fields[snr_field]['data']
+    else:
+        raise KeyError('Field not available: ' + snr_field)
+    if zdr_field in radar.fields:
+        zdrdB = radar.fields[zdr_field]['data']
+    else:
+        raise KeyError('Field not available: ' + zdr_field)
+    if nh_field in radar.fields:
+        nh = radar.fields[nh_field]['data']
+    else:
+        raise KeyError('Field not available: ' + nh_field)
+    if nv_field in radar.fields:
+        nv = radar.fields[nv_field]['data']
+    else:
+        raise KeyError('Field not available: ' + nv_field)
+
+    snr_h = np.ma.power(10., 0.1*snrdB_h)
+    zdr = np.ma.power(10., 0.1*zdrdB)
+    alpha = np.ma.power(10., 0.1*(nh-nv))
+
+    mask = np.ma.getmaskarray(urhohv)
+    fill_value = urhohv.get_fill_value()
+
+    rhohv_data = urhohv*np.ma.sqrt((1.+1./snr_h)*(1.+zdr/(alpha*snr_h)))
+    rhohv_data[rhohv_data > 1.] = 1.
+    rhohv_data.set_fill_value(fill_value)
+    rhohv_data.data[mask.nonzero()] = fill_value
```

Comment:
This should be `rhohv_data.data[mask] = np.ma.masked`  Additional I do not think setting the fill_value is really needed.